### PR TITLE
chore: reorganize repo for multiple packages

### PR DESCRIPTION
**Pull request by federicobond**
This is a ~work-in-progress~ proposal for organizing the repo around multiple packages

### CODE REVIEW

**Comment by federicobond** [comment]
Tooling is still a bit of a hack, happy to take suggestions. I mostly wanted to get the basic structure right.

**Review by beeme1mr — APPROVED** [review]

**Comment by federicobond** [comment]
Ok, so I did some research on the tooling for packaging and dependency management and after looking at different options, including Flit, Poetry and PDM, I have integrated a tool called [Hatch](https://hatch.pypa.io/) that is being developed under the umbrella of the PyPA.

Hatch solves the problem of running tests in isolated environments for each package, which we weren't able to do with the existing Makefile setup. It's easy to understand, fast, and adheres much better to modern Python packaging standards (Poetry is known for doing its own thing, sometimes with questionable choices for package authors).

The only downside of Hatch is that there is no explicit monorepo support yet, but I don't believe it to be a deal breaker at the moment. 

I think we can give it a try and see if works for our use case. Worst case scenario is we have to tweak up pyproject.toml files to use a different tool.

**Comment by beeme1mr** [comment]
I'll look into this more tomorrow but it looks like great progress.

**Comment by beeme1mr** [comment]
@federicobond could you please sign off on the commits when you have a moment?

**Comment by federicobond** [comment]
Sure! I was missing the last one apparently, sorry about that.

**Review by gruebel — APPROVED** [review]
nice work 🍻

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -22,6 +22,9 @@ jobs:
     strategy:
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        package:
+          - "hooks/openfeature-hooks-opentelemetry"
+          - "providers/openfeature-provider-flagd"
```
[the remark this answers is not in the document]
Comment (federicobond) [reply .github/workflows/build.yml:27]: Good point, I think we should err on the side of building every package if there is any change that might affect the entire repo, the builds parallelize quite easily in GH Actions, but we can definitely look into filtering as an optimization

**Pull request merged**